### ISSUE TITLE
linter: `promise/no-return-wrap` fails to detect the violation in conditionals

### ISSUE THREAD

**Issue by JasonKaz** [oxc-project/oxc#22469]
### What version of Oxlint are you using?

1.64.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["promise"],
  "rules": {
    "promise/no-return-wrap": "error"
  }
}
```

### What happened?

oxlint only detects Promise.reject/resolve wrapping at the top level of a `.then()` callback body. Returns nested inside conditionals are not flagged, unlike eslint-plugin-promise which uses full code-path analysis.

Demo can be found [here](https://playground.oxc.rs/?lintRules=promise%2Fno-return-wrap&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22promise%2Fno-return-wrap%5C%22%3A%5C%22error%5C%22%7D%2C%5C%22plugins%5C%22%3A%5B%5C%22eslint%5C%22%2C%5C%22oxc%5C%22%2C%5C%22typescript%5C%22%2C%5C%22unicorn%5C%22%2C%5C%22promise%5C%22%5D%7D%22%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=%2F**%0A+*+oxlint+only+detects+Promise.reject%2Fresolve+wrapping+at+the+top+level+of+a%0A+*+.then%28%29+callback+body.+Returns+nested+inside+conditionals+are+not+flagged%2C%0A+*+unlike+eslint-plugin-promise+which+uses+full+code-path+analysis.%0A+*%2F%0A%0Adeclare+const+condition%3A+boolean%3B%0A%0A%2F%2F+%E2%9C%85+DETECTED+%E2%80%94+top-level+return+Promise.reject+inside+.then%28%29%0APromise.resolve%28%29.then%28%28%29+%3D%3E+%7B%0A++++return+Promise.reject%28new+Error%28%27top-level+reject%27%29%29%3B%0A%7D%29%3B%0A%0A%2F%2F+%E2%9C%85+DETECTED+%E2%80%94+top-level+return+Promise.resolve+inside+.then%28%29%0APromise.resolve%28%29.then%28%28%29+%3D%3E+%7B%0A++++return+Promise.resolve%28%27top-level+resolve%27%29%3B%0A%7D%29%3B%0A%0A%2F%2F+%E2%9C%85+DETECTED+%E2%80%94+top-level+return+Promise.reject+inside+.catch%28%29%0APromise.resolve%28%29.catch%28%28%29+%3D%3E+%7B%0A++++return+Promise.reject%28new+Error%28%27top-level+reject+in+catch%27%29%29%3B%0A%7D%29%3B%0A%0A%2F%2F+%E2%9D%8C+NOT+DETECTED+%E2%80%94+return+Promise.reject+nested+inside+an+if+block%0APromise.resolve%28%29.then%28%28%29+%3D%3E+%7B%0A++++if+%28condition%29+%7B%0A++++++++return+Promise.reject%28new+Error%28%27nested+reject%27%29%29%3B%0A++++%7D%0A%7D%29%3B%0A%0A%2F%2F+%E2%9D%8C+NOT+DETECTED+%E2%80%94+return+Promise.resolve+nested+inside+an+if+block%0APromise.resolve%28%29.then%28%28%29+%3D%3E+%7B%0A++++if+%28condition%29+%7B%0A++++++++return+Promise.resolve%28%27nested+resolve%27%29%3B%0A++++%7D%0A%7D%29%3B).
